### ISSUE TITLE
Incorrect Results when filtering by Label

### ISSUE THREAD

**Issue by pvtpogopuschel** [RedisTimeSeries/RedisTimeSeries#995]
Hey, dear Redis Time Series Team!

We are running a fairly large data set using Redis Time Series and have experienced an occasional issue with filtering by labels using MGET or MRANGE. This happens - at the moment - with around 10% of our queries, but the rate seems to be growing. 

Let us have a look at this example.

```
>> TS.INFO testSeries:61433f8cb935aa00087b8a37
 1) "totalSamples"
 2) (integer) 5984
 3) "memoryUsage"
 4) (integer) 46032
 5) "firstTimestamp"
 6) (integer) 1632484800000
 7) "lastTimestamp"
 8) (integer) 1642171800000
 9) "retentionTime"
10) (integer) 0
11) "chunkCount"
12) (integer) 11
13) "chunkSize"
14) (integer) 4096
15) "chunkType"
16) "compressed"
17) "duplicatePolicy"
18) (nil)
19) "labels"
20) 1) 1) "5bb895451f355bb5735d0f40e2aa90dd"
       2) "helloworld"
21) "sourceKey"
22) (nil)
23) "rules"
24) (empty list or set)
```

As you can see, time series `testSeries:61433f8cb935aa00087b8a37` has a label `5bb895451f355bb5735d0f40e2aa90dd` with the value `helloworld`. Let us run `MGET` for it:

```
TS.MGET WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=helloworld
(empty list or set)
```

The result is empty. Let us verify that the time series does have data:

```
TS.GET testSeries:61433f8cb935aa00087b8a37
1) (integer) 1642171800000
2) "37.945"
```

We are currently running redis time series 1.4.10, the changelog does not seem to indicate that this would be solved with 1.4.13. Have you experienced a similar issue?

**Comment by gkorland** [comment]
- Do you run in a cluster or single shard? 

- Did you try with 1.6?

**Comment by WeaselScience** [comment]
* We are running a replicated setup with Sentinel, consisting of three nodes.
* We did not try with 1.6, I'll post findings as soon as we do.

Some more details:
* This issue is present in all three of our replicas, i.e. `TS.INFO` for a given key indicates that the label exists on all three replicas, and on all three replicas `TS.MGET WITHLABELS` returns an empty result.

**Comment by WeaselScience** [comment]
We've found a way to reproduce with just one timeseries. We have dumped a broken key using this command: `./redis-cli --raw dump energyDaySum:61433f8cb935aa00087b8a37 | head -c-1 > bugged-timeseries-dump`.

To reproduce:
* Place `repro.sh` and `bugged-timeseries-dump` ([Archive.zip](https://github.com/RedisTimeSeries/RedisTimeSeries/files/7888827/Archive.zip)) in a directory.
* Run:
```
docker run -it --rm --mount type=bind,src=$(pwd),dst=/test --workdir /test redislabs/redistimeseries:1.4.10 bash repro.sh
```

Our observations:
* This issue is reproducible in `1.4.10`, `1.4.13` and `1.6.0`, but is NOT reproducible from `1.6.1` onwards.
* The imported timeseries has the label we expect it to have from the dump.
* The timeseries is not part of the result when queried with `TS.MGET WITHLABELS`.
* An error is printed in the Redis console upon running `TS.MGET WIHTLABELS`:
```
# <timeseries> couldn't open key or key is not a Timeseries. key=daySum:61433f8cb935aa00087b8a37
```
* The above error seems to indicate that `TS.MGET WITHLABELS` is attempting to access a timeseries key that no longer exists. The key in the error is in fact an old name of the key in question, as it was before we ran a migration and RENAMEd the key from `daySum:61433f8cb935aa00087b8a37` to `energyDaySum:61433f8cb935aa00087b8a37`

Full repro.sh output:
```
+ redis-server --daemonize yes --loadmodule /usr/lib/redis/modules/redistimeseries.so --logfile /redis-log
+ cat bugged-timeseries-dump
+ redis-cli -x restore energyDaySum:61433f8cb935aa00087b8a37 0
OK
+ redis-cli ts.info energyDaySum:61433f8cb935aa00087b8a37
 1) totalSamples
 2) (integer) 6149
 3) memoryUsage
 4) (integer) 50208
 5) firstTimestamp
 6) (integer) 1632484800000
 7) lastTimestamp
 8) (integer) 1642503000000
 9) retentionTime
10) (integer) 0
11) chunkCount
12) (integer) 12
13) chunkSize
14) (integer) 4096
15) chunkType
16) compressed
17) duplicatePolicy
18) (nil)
19) labels
20) 1) 1) "5bb895451f355bb5735d0f40e2aa90dd"
       2) "daySum"
21) sourceKey
22) (nil)
23) rules
24) (empty array)
+ redis-cli TS.MGET WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=daySum
(empty array)
+ cat /redis-log
10:C 18 Jan 2022 11:52:16.966 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
10:C 18 Jan 2022 11:52:16.967 # Redis version=6.0.9, bits=64, commit=00000000, modified=0, pid=10, just started
10:C 18 Jan 2022 11:52:16.967 # Configuration loaded
10:M 18 Jan 2022 11:52:16.972 * Running mode=standalone, port=6379.
10:M 18 Jan 2022 11:52:16.972 # Server initialized
10:M 18 Jan 2022 11:52:16.972 # WARNING overcommit_memory is set to 0! Background save may fail under low memory condition. To fix this issue add 'vm.overcommit_memory = 1' to /etc/sysctl.conf and then reboot or run the command 'sysctl vm.overcommit_memory=1' for this to take effect.
10:M 18 Jan 2022 11:52:16.973 * <timeseries> RedisTimeSeries version 10410, git_sha=d7ccf2a41cb2fdc92d00f3cc930477467e661916
10:M 18 Jan 2022 11:52:16.973 * <timeseries> Redis version found by RedisTimeSeries : 6.0.9 - oss
10:M 18 Jan 2022 11:52:16.974 * Module 'timeseries' loaded from /usr/lib/redis/modules/redistimeseries.so
10:M 18 Jan 2022 11:52:16.975 * Ready to accept connections
10:M 18 Jan 2022 11:52:16.988 # <timeseries> couldn't open key or key is not a Timeseries. key=daySum:61433f8cb935aa00087b8a37
```

**Comment by WeaselScience** [comment]
Hello once again, we've found more issues that are present in RedisTimeSeries `1.6.7` aswell. We found that `TS.MGET WITHLABELS` and `TS.MRANGE WITHLABELS` return different timeseries name. The above reproduction can reproduce this issue with an updated `repro.sh` and same timeseries dump.

<details>
<summary>repro.sh</summary>

```
set -x
redis-server --daemonize yes --appendonly yes --save "" --loadmodule /usr/lib/redis/modules/redistimeseries.so --logfile /redis-log --dir /test
sleep 1
cat bugged-timeseries-dump | redis-cli -x restore energyDaySum:61433f8cb935aa00087b8a37 0
redis-cli keys "*"
redis-cli ts.info energyDaySum:61433f8cb935aa00087b8a37
redis-cli TS.MGET WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=daySum
redis-cli TS.MRANGE 1642644000000 1642644000000 WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=daySum
cat /redis-log
```
</details>

Here is a rundown of steps as executed using above-described reproduction (after the dump was imported):

`redis-cli keys '*'`
```
1) "energyDaySum:61433f8cb935aa00087b8a37"
```
We confirm that the data import imports only one key.

---

`redis-cli ts.info energyDaySum:61433f8cb935aa00087b8a37`
```
 1) totalSamples
 2) (integer) 6149
 3) memoryUsage
 4) (integer) 50208
 5) firstTimestamp
 6) (integer) 1632484800000
 7) lastTimestamp
 8) (integer) 1642503000000
 9) retentionTime
10) (integer) 0
11) chunkCount
12) (integer) 12
13) chunkSize
14) (integer) 4096
15) chunkType
16) compressed
17) duplicatePolicy
18) (nil)
19) labels
20) 1) 1) "5bb895451f355bb5735d0f40e2aa90dd"
       2) "daySum"
21) sourceKey
22) (nil)
23) rules
24) (empty array)
```
We confirm that the imported timeseries has a label `5bb895451f355bb5735d0f40e2aa90dd=daySum`

---

`redis-cli TS.MGET WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=daySum`
```
1) 1) "energyDaySum:61433f8cb935aa00087b8a37"
   2) 1) 1) "5bb895451f355bb5735d0f40e2aa90dd"
         2) "daySum"
   3) 1) (integer) 1642503000000
      2) 19.296999999999997
```
We confirm that running `TS.MGET WITHLABELS` returns the correct timeseries name of `energyDaySum:61433f8cb935aa00087b8a37`.

---

`redis-cli TS.MRANGE 1642644000000 1642644000000 WITHLABELS FILTER 5bb895451f355bb5735d0f40e2aa90dd=daySum`
```
1) 1) "daySum:61433f8cb935aa00087b8a37"
   2) 1) 1) "5bb895451f355bb5735d0f40e2aa90dd"
         2) "daySum"
   3) (empty array)
```
We witness the issue: `TS.MRANGE` returns an incorrect timeseries name of `daySum:61433f8cb935aa00087b8a37`, a timeseries that does not even exist in the Redis instance.

Again, the timeseries from the dump was at some point called `daySum:61433f8cb935aa00087b8a37`, but was renamed to `energyDaySum:61433f8cb935aa00087b8a37` at some point. 

As this issue is still reproducible on RTS `1.6.7`, I'd like to keep this ticket open. What would be the course of action for us? We've already tried running `TS.ALTER` to remove and re-add the label, but the issue persists. Is our database actually corrupted? If so, do you have any suggestion on how we could uncorrupt it?

Thank you very much for your time.

**Comment by OfirMos** [comment]
Hello, we will soon have a PR which fix this.
In the mean time an easy workaround will be to rename energyDaySum:61433f8cb935aa00087b8a37 to different temporary name and then rename it back to energyDaySum:61433f8cb935aa00087b8a37.
For example:
RENAME energyDaySum:61433f8cb935aa00087b8a37 energyDaySum:61433f8cb935aa00087b8a37_temp
RENAME energyDaySum:61433f8cb935aa00087b8a37_temp energyDaySum:61433f8cb935aa00087b8a37

Please let us know if it worked for you.

**Comment by OfirMos** [comment]
Hi, the bug was fixed in v1.6.8

**Comment by WeaselScience** [comment]
Thank you, we'll upgrade our version and post an update.
